tests/integration/test_task.py

```diff
@@ -80,3 +83,229 @@ def _run(self, input_files):
+    task = list(tasks)[0]
+    assert task.products == set()
+    assert task.output_files is None
+    assert task.activity is None
+    assert task.priority == 0
+
+
+def _get_single_diagnostic_task(tmp_path, diag_script, write_diag=True):
+    """Assemble a simple DiagnosticTask object."""
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_settings = {'run_dir': diag_run_dir, 'profile_diagnostic': False}
+    if write_diag:
+        with open(diag_script, "w") as fil:
+            fil.write("import os\n\nprint(os.getcwd())")
+
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=[BaseTask(name=f'task0-ancestor{j}') for j in range(2)],
+        script=diag_script,
+        settings=diag_settings,
+        output_dir=diag_output_dir,
+    )
+
+    return task
+
+
+def test_py_diagnostic_task_constructor(tmp_path):
+    """Test DiagnosticTask basic attributes."""
+    diag_script = tmp_path / 'diag_cow.py'
+    task = _get_single_diagnostic_task(tmp_path, diag_script)
+    assert task.name == 'task0'
+    ancestor_names = [anc.name for anc in task.ancestors]
+    assert ancestor_names == ['task0-ancestor0', 'task0-ancestor1']
+    assert task.script == diag_script
+    assert task.settings == {
+        'run_dir': tmp_path / 'mydiag' / 'run_dir',
+        'profile_diagnostic': False
+    }
+    assert task.output_dir == tmp_path / 'mydiag'
+
+
+def test_diagnostic_diag_script_none(tmp_path):
+    """Test case when diagnostic script doesnt exist."""
+    diag_script = tmp_path / 'diag_cow.py'
+    with pytest.raises(DiagnosticError) as err_msg:
+        _get_single_diagnostic_task(tmp_path, diag_script, write_diag=False)
+    diagnostics_root = os.path.join(DIAGNOSTICS_PATH, 'diag_scripts')
+    script_file = os.path.abspath(os.path.join(diagnostics_root, diag_script))
+    ept = ("Cannot execute script '{}' "
+           "({}): file does not exist.".format(script_file, script_file))
+    assert ept == str(err_msg.value)
+
+
+def _get_diagnostic_tasks(tmp_path, diagnostic_text, extension):
+    """Assemble Python diagnostic tasks of DiagnosticTasks."""
+    diag = 'diag_cow.' + extension
+    diag_script = tmp_path / diag
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_plot_dir = diag_output_dir / 'plot_dir'
+    diag_work_dir = diag_output_dir / 'work_dir'
+    diag_settings = {
+        'run_dir': diag_run_dir.as_posix(),
+        'plot_dir': diag_plot_dir.as_posix(),
+        'work_dir': diag_work_dir.as_posix(),
+        'profile_diagnostic': False,
+        'exit_on_ncl_warning': False
+    }
+
+    with open(diag_script, "w") as fil:
+        fil.write(diagnostic_text)
+
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=None,
+        script=diag_script.as_posix(),
+        settings=diag_settings,
+        output_dir=diag_output_dir.as_posix(),
+    )
+
+    return {task}
```

Comment:
Why are you making a set with this task in it? In the tests below you only use the task and need extra code again to take it out of the set

Reply:
:facepalm: